gluten-flink/planner/src/main/java/org/apache/gluten/rexnode/Utils.java

```diff
@@ -46,4 +56,42 @@ public static void registerRegistry() {
             ISerializableRegistry.registerAll();
         }
     }
+
+    private static final List<Type> NUMBER_TYPE_PRIORITY_LIST = List.of(
+        new TinyIntType(),
+        new SmallIntType(),
+        new IntegerType(),
+        new BigIntType(),
+        new RealType(),
+        new DoubleType()
+    );
+
+    private static int getNumberTypePriority(Type type) {
+        int index = -1;
+        for (int i = 0; i < NUMBER_TYPE_PRIORITY_LIST.size(); ++i) {
+            if (NUMBER_TYPE_PRIORITY_LIST.get(i).getClass().equals((type.getClass()))) {
+                index = i;
+                break;
+            }
+        }
+        if (index == -1) {
```

Comment:
No need to use index as flag. Just throw exception here as there is no matching in the loop.

Reply:
done